### PR TITLE
Add requests as a legit dependency

### PR DESCRIPTION
**Pull request by jamesr66a**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25596 Add requests as a legit dependency**

Giving up on trying to limit this to just a py2 dependency

Differential Revision: [D17171063](https://our.internmc.facebook.com/intern/diff/D17171063)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,3 +4,4 @@ pyyaml
 setuptools
 six
 typing
+requests
```
Comment (pietern) [inline requirements.txt]: Seems to be ordered alphabetically.

**Review by pietern — COMMENTED** [review]
LGTM, but will defer approval to @ezyang or @soumith since this adds a new dep.

**Comment by soumith** [comment]
but why?

**Comment by jamesr66a** [comment]
@soumith because this seems like a pretty poor experience: https://github.com/pytorch/pytorch/pull/25083#issuecomment-527219928

**Review by ezyang — APPROVED** [review]

File: torch/hub.py [hunk 2]
```diff
@@ -374,24 +368,11 @@ def load(github, model, *args, **kwargs):
...
 
-        meta = u.info()
-        if hasattr(meta, 'getheaders'):
-            content_length = meta.getheaders("Content-Length")
-        else:
-            content_length = meta.get_all("Content-Length")
-        if content_length is not None and len(content_length) > 0:
-            file_size = int(content_length[0])
+    content_length = response.headers['Content-Length']
+    file_size = content_length
```
Comment (xuhdev) [inline torch/hub.py]: Actually I don't see how `content_length` is used. Perhaps we should remove it?

**Review by soumith — CHANGES_REQUESTED** [review]
can you please add requests to requirements.txt and to README but leave the code unchanged.
Once we switch to Python3 when Py2 EOLs, we can remove this dependency with the current code.

**Comment by jamesr66a** [comment]
@soumith does this look OK now?

**Review by soumith — APPROVED** [review]
thanks lgtm

**Pull request closed without merging**